### PR TITLE
[build] Unify logic to skip building projects in build-script-impl

### PR DESCRIPTION
**Pull request by bulbazord**
cc @compnerd @Rostepher

### CODE REVIEW

**Comment by xiaobai** [comment]
@swift-ci please test

File: utils/build-script-impl [hunk 1]
```diff
@@ -2228,48 +2230,46 @@ for host in "${ALL_HOSTS[@]}"; do
...
+                            LIB_NAME="libclang_rt.$OS.a"
+                            HOST_LIB_PATH="$HOST_CXX_BUILTINS_DIR/$LIB_NAME"
+                            if [[ -f "${HOST_LIB_PATH}" ]]; then
+                                call cp "${HOST_LIB_PATH}" "${DEST_BUILTINS_DIR}/${LIB_NAME}"
+                            elif [[ "${VERBOSE_BUILD}" ]]; then
+                                echo "no file exists at ${HOST_LIB_PATH}"
+                            fi
                         done
-                    fi
+                    done
```
Comment (xiaobai) [inline utils/build-script-impl]: This whole thing is kinda ugly but I just removed the outer `if` condition that use the `skip_build` variable.

**Review by compnerd — APPROVED** [review]

File: utils/build-script-impl (hunk 1 above)
Comment (compnerd) [reply utils/build-script-impl]: Yeah, its all whitespace diff really.  Hint: add `?w=1` to the diff URL to get the real diff

**Comment by compnerd** [comment]
@swift-ci please smoke test Linux platform

**Comment by xiaobai** [comment]
@swift-ci please test Linux platform

**Pull request merged**